test/e2e/e2e_helpers_test.go

```diff
@@ -424,6 +423,65 @@ func sortNamespacedNames(nns []types.NamespacedName) {
 }
 
 func testDataPath(description ginkgo.GinkgoTestDescription) string {
-       path := filepath.Join(description.ComponentTexts...)
-       return strings.ReplaceAll(path, " ", "_")
+	path := filepath.Join(description.ComponentTexts...)
+	return strings.ReplaceAll(path, " ", "_")
+}
+
+func GetIstioObject(istio *istiov1beta1.Istio, namespace, name string) error {
+	return testEnv.Client.Get(context.TODO(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      name},
+		istio)
+}
+
+func GetMixerlessTelemetryStatus(istio *istiov1beta1.Istio, namespace, name string) (string, error) {
+
+	err := GetIstioObject(istio, namespace, name)
+	if err != nil {
+		return "", err
+	}
+	// query current state and return it
+	if istio.Spec.MixerlessTelemetry.Enabled == nil {
+		return "N", nil
+	}
+	if *istio.Spec.MixerlessTelemetry.Enabled {
+		return "T", nil
+	}
+	return "F", nil
+}
+
+func SetMixerlessTelemetryState(istio *istiov1beta1.Istio, namespace, name, newState string) (bool, error) {
+	var expectMissingFilter bool
+	err := GetIstioObject(istio, namespace, name)
+	if err != nil {
+		return false, err
+	}
+	switch newState {
+	case "T": // transition to true
+		istio.Spec.MixerlessTelemetry.Enabled = pkgutil.BoolPointer(true)
+		expectMissingFilter = false
+	case "F": // transition to false
+		istio.Spec.MixerlessTelemetry.Enabled = pkgutil.BoolPointer(false)
+		expectMissingFilter = true
+	case "N": // transition to nil
+		istio.Spec.MixerlessTelemetry.Enabled = nil
+		expectMissingFilter = true
+	}
+	return expectMissingFilter, nil
+}
```

Comment:
I think I have resolved all of these problems.
Get/Set MixerlessTelemetryStatus are more symmetrical and don't return the magic expectMissingFilter.
No switch is needed since we can directly set nil or use BoolPointer.